### PR TITLE
Document non-reproducible outputs

### PR DESCRIPTION
**Pull request by aherrmann-da**
The frontend webpack bundle and the vscode extension are non-reproducible outputs in the SDK release tarball. I looked into what makes them non-reproducible. I didn't succeed in making them reproducible, but this PR adds my findings as comments to the relevant Bazel targets.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/daml-extension/BUILD.bazel [hunk 1]
```diff
@@ -73,6 +73,10 @@ genrule(
     tools = ["//bazel_tools/sh:mktgz"],
 )
 
+# This rule is not reproducible. `vsce package` generates a `.vsix` file which
+# is just a zip archive. The order of entries in that archive is
+# non-deterministic and the individual entries contain non-reproducible
+# timestamps.
```
Comment (cocreature) [inline compiler/daml-extension/BUILD.bazel:79]: Could we unzip and rezip to fix this?

File: navigator/frontend/BUILD.bazel [hunk 2]
```diff
@@ -64,6 +64,15 @@ nodejs_binary(
 
 # Builds the frontend single page application and bundles all output files
 #  (HTML + JavaScript + CSS + images) in a .tgz or .jar archive.
+# This target is non-reproducible. The browsercheck and bundle outputs
```
Comment (cocreature) [inline navigator/frontend/BUILD.bazel:67]: Non-reproducible hashes sound very weird :confused:

File: compiler/daml-extension/BUILD.bazel (hunk 1 above)
Comment (aherrmann-da) [reply compiler/daml-extension/BUILD.bazel:79]: Yes, I was hoping to find a way to do this with vsce directly, but failing that this should do it.

Comment (cocreature) [reply compiler/daml-extension/BUILD.bazel:79]: Not much to be gained trying to fix this if we cannot fix Navigator. In the end, the thing we really want to be reproducible is the SDK tarball.

**Pull request merged**